Ez leka/readiness probe

ez-leka opened the pull request:
1. Implemented readiness probe to make sure when pod says ready it has completed all installations, including KKM which takes a long time
2. Implemented daemonset to label node  it has run on as kontain ready. All kontain applications should have 2 things added 
runrimeClass and 
nodeSelector:
        sandbox: kontain
to ensure that they are assigned to the properly configured node because runtimeClass creation is cluster-wide

ez-leka committed:
implemented readiness probe and node labeling from withing daemonset

ez-leka committed:
service account to allow for node labeling

ez-leka committed:
added check for  successful node labeling

gnode1 commented on helpers/minikube-cluster.sh:
```diff
@@ -75,7 +75,7 @@ create_cluster() {
     fi
     
-    minikube start --container-runtime="$runtime" --driver="$driver" "$extra_options" --wait=all --logtostderr=true  --wait-timeout=10m0s
+    minikube start --container-runtime="$runtime" --driver="$driver" "$extra_options" --wait=all --logtostderr=true  -v=10 --wait-timeout=10m0s
```
what is `-v=10`?

gnode1 commented on tests/test.yaml:
```diff
@@ -12,8 +12,10 @@ spec:
-        image: busybox:latest
-        command: [ "sleep", "infinity" ]
-        imagePullPolicy: IfNotPresent
+        - name: kontain-test-app
```
For my information - how did ti work before?

ez-leka committed:
gke and AKS clusters MUST have different name. Otherwse GKE fails with
kube-dns not starting

Please enter the commit message for your changes. Lines starting

gnode1 commented on .github/workflows/ci-workflow.yaml:
```diff
@@ -203,6 +203,7 @@ jobs:
   test-gce:
     name: GCE Deployment
     runs-on: ubuntu-20.04
+    #needs: [test-aks]
```
remove?

ez-leka committed:
force systemd in minikube

ez-leka committed:
removed some extra comments

gnode1 reviewed (approved):
LGTM with a couple of nits
